src/ceph-disk/ceph_disk/main.py

```diff
@@ -2623,6 +2623,9 @@ def __init__(self, args):
         if self.args.osd_uuid is None:
             self.args.osd_uuid = str(uuid.uuid4())
 
+        if os.path.realpath(self.args.data).startswith('/dev/'):
+            self.args.data_dev = True
+
```

Comment:
> Added test for when dir is a symlink to somewhere in /dev

where is this test?

> Moved tests so that an argument with a real path that starts with /dev/ will automatically invoke --data-dev

this is misleading, you didn't *move* tests. you are using `--data-dev` to fail the test if the provided argument starts with /dev is not a block device. see http://tracker.ceph.com/issues/18976.